### PR TITLE
gpui: Update Menu name to use `SharedString` type to support more types

### PR DESCRIPTION
**Pull request by huacnlee**
Release Notes:

- N/A

### CODE REVIEW

**Comment by huacnlee** [comment]
Sorry, I meant to submit a post on my fork first, then create a discussion to see if this suggestion is accepted, but I accidentally created it here. 

Let's discuss it here.

**Comment by huacnlee** [comment]
I want to change the name type of Menu because I tried to add I18n support to Zed before, and used the rust-i18n crate that I implemented. The `t!` method in it needs to support runtime strings (such as `format!("Close all {} items", 5)`), so the return value is of type Cow instead of static's str.

Here is a draft:

https://github.com/huacnlee/zed/pull/3

Or sometime, the Menu name may be a dynamic string, so I want to change the type of Menu name to Cow.

File: crates/gpui/src/platform/app_menu.rs [hunk 1]
```diff
@@ -6,7 +6,7 @@ use util::ResultExt;
 /// A menu of the application, either a main menu or a submenu
 pub struct Menu<'a> {
     /// The name of the menu
-    pub name: &'a str,
+    pub name: Cow<'a, str>,
```
Comment (maxdeviant) [inline crates/gpui/src/platform/app_menu.rs:9]: Have we considered using a `SharedString` here?

That type is more commonly used in GPUI.

Comment (huacnlee) [reply crates/gpui/src/platform/app_menu.rs]: `SharedString` is ok, it can handle same things.

**Review by maxdeviant — APPROVED** [review]
Thanks, this seems like a useful thing to have!

**Comment by huacnlee** [comment]
@maxdeviant And I have almost done the I18n support for Zed.

https://github.com/huacnlee/zed/pull/3

Are there any plans to introduce this support at this stage? After the introduction, we need to keep using `t!()` instead of static str for each display string. But it is simple, I also did backward compatibility. If some translate is missing, the original string will be used for display, which will not have much impact on usage.

This may increase the workload of Zed development.

**Comment by maxdeviant** [comment]
> Are there any plans to introduce this support at this stage?

We haven't really discussed it internally yet, but my guess is that it's probably a ways off.

**Comment by huacnlee** [comment]
Okay, I'll put it aside for now.

**Pull request merged**